assignment_4.py

```diff
@@ -0,0 +1,63 @@
+
+def count_islands(rows, columns, grid):
+    '''
+        INPUT - # of rows
+              - # of columns
+              - 2-dimensional array of booleans
+        RETURN - number of islands      
+    '''
+    
+    visited = [[False for i in range(columns)] for i in range(rows)]
+    count  = 0
+    for row in range(rows):
+        for col in range(columns):
+            if grid[row][col] == 'T' and not visited[row][col]:
+                check_tile(rows, columns, row, col, grid, visited)
+                count += 1
+    
+
+    return count
+        
+def check_tile(rows, columns, x, y, grid, visited):
+    '''
+    INPUT - # of rows
+          - # of columns
+          - position x, y on the grid
+          - 2-dimensional array of booleans (grid)
+          - 2-dimensional array of visited positions

```

Comment:
Please make clear that the "visited" parameter will be modified by the function and will actually contain the result of the function's elaboration.